_posts/2016-12-05-haskell-package-checklist.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Haskell package checklist
+---
+
+- Host on GitHub
+  - Easy for other developers to contribute.
+  - Integrates well with other services.
+  - More popular than other hosts.
+- Build with Stack
+  - Easiest way to manage dependencies.
+  - Ensures that your package will remain buildable.
+  - Makes difficult workflows easy, like `--profile` and `--coverage`.
+- Use `hpack`
+  - Uses YAML instead of custom package format.
+  - Avoids unnecessary boilerplate.
+  - Integrated directly into Stack.
+- Name package with `kebab-case`
+  - Keep everything lowercase to avoid confusion.
+  - Use hyphens to separate words.
+  - Keep it short and memorable.
+- Use Semantic Versioning
+  - SemVer is used by many other languages.
+  - It matches how developers generally think about versions.
+  - Hackage recommends the PVP, which adds ambiguity.
+- Choose a license
+  - Packages without a license can’t really be used.
+  - BSD-3-Clause is the most popular, followed by MIT.
+  - Include the license file as well.
+- Write a README
```

Comment:
Good points! I usually end up with `README.markdown`, `CHANGELOG.markdown`, and `LICENSE.markdown`. All in `extra-source-files` as you suggest.